BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -233,9 +256,20 @@ actor DefaultAuthenticatorSyncService: NSObject, AuthenticatorSyncService {
     ///   - userId: The userId of the account to which the Ciphers belong.
     ///
     private func writeCiphers(ciphers: [Cipher], userId: String) async throws {
-        guard !vaultTimeoutService.isLocked(userId: userId) else { return }
+        let account = try await stateService.getAccount(userId: userId)
+        let useKey = vaultTimeoutService.isLocked(userId: userId)
 
-        let items = try await decryptTOTPs(ciphers, userId: userId)
-        try await authBridgeItemService.replaceAllItems(with: items, forUserId: userId)
+        do {
+            if useKey {
+                try await authRepository.unlockVaultWithAuthenticatorVaultKey(userId: userId)
+            }
+            let items = try await decryptTOTPs(ciphers, account: account)
+            try await authBridgeItemService.replaceAllItems(with: items, forUserId: userId)
+        } catch {
+            errorReporter.log(error: error)
+        }
+        if useKey {
+            await vaultTimeoutService.lockVault(userId: userId)
+        }
```

Comment:
⛏️ Maybe I couldn't find it, but could a test be added when `let items = try await decryptTOTPs(ciphers, account: account)` throws? and when `try await authBridgeItemService.replaceAllItems(with: items, forUserId: userId)` throws ensuring the vault is locked afterwards?

Reply:
Added two more tests to explicitly check each of those 2 lines.  👍  Verifies that the vault gets properly re-locked even when we hit an error.